catalog/imaging/imaging-postprocess-plugin-nitf/src/test/groovy/org/codice/alliance/plugin/nitf/NitfPostProcessPluginSpec.groovy

```diff
@@ -0,0 +1,406 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.plugin.nitf
+
+import ddf.catalog.data.Attribute
+import ddf.catalog.data.Metacard
+import ddf.catalog.data.types.Core
+import org.codice.ddf.catalog.async.data.api.internal.ProcessCreateItem
+import org.codice.ddf.catalog.async.data.api.internal.ProcessRequest
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource
+import org.codice.ddf.catalog.async.data.api.internal.ProcessUpdateItem
+import org.codice.ddf.catalog.async.data.impl.ProcessRequestImpl
+import org.codice.imaging.nitf.core.image.ImageSegment
+import org.codice.imaging.nitf.render.NitfRenderer
+import spock.lang.Ignore
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.awt.image.BufferedImage
+import java.util.function.Supplier
+
+class NitfPostProcessPluginSpec extends Specification {
```

Comment:
❗️We need tests for the overview resizing to make sure `calculateOverviewHeight()` and `calculateOverviewWidth()` as expected.

Reply:
Good catch, I'll add those now

Reply:
So I had tried to add some tests to these but was having trouble getting the mocks for the `NitfParserInputFlow` and `NitfParserParsingFlow` to throw exceptions (or actually do anything at all). It seems like they don't respect the behavior I've given them and I end up getting real things back from them instead... Any ideas as to why that would happen @lessarderic?

Reply:
❓Did we add some tests to cover the `calculateOverviewHeight()` and `calculateOverviewWidth()` methods? I'm still worried about the logic in there.

Reply:
I'll investigate the mocking problem and comment if I find something.

Reply:
I was able to get `"encountering nitf formatting errors during nitf parsing in create requests"` to pass by replacing
```
NitfParserInputFlowImpl nitfParserInputFlow = Mock(NitfParserInputFlowImpl)
```
with
```
NitfParserInputFlow nitfParserInputFlow = Mock(NitfParserInputFlow)
```

Reply:
Ok, looks like the two other ignored tests are failing for two reasons.

First, the methods should call `multiNitfInteractions()` in a `setup:` block, not a `when:` block, otherwise the mocks aren't properly initialized when the `then:` block is run (which happens _before_ the `when:` is run). So the method should look like this:

```
        setup:
            interaction {
                multiNitfInteractions()
            }
        when: "process request contains multiple nitf resources"
            nitfPostProcessPlugin.processCreate(createProcessRequest)
        then: "resources should be added to their respective metacards"
            ...
 ```

The other issue is that we create the `createProcessRequest` and `updateProcessRequest` in the `setup()` method and configure `getProcessItems()` to return a single item, and then add another `getProcessItem()` expectation in `multiNitfInteractions`. However, doing does _not_ override the expectation in `setup()`, but rather tells Spock what to return on a second call to `getProcessItem()`.

Reply:
Awesome that fixed everything. Thanks @lessarderic!